rrrspec-client/lib/rrrspec/registry.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Registry
+
+    @@runners = Hash.new
+
+    def self.reset_runners
+      @@runners = []
+    end
+
+    def self.register_runner(factory, file_extension)
+      @@runners[file_extension] = factory
+    end
+
+
+    def self.get_runner(file_extension)
+      @@runners[file_extension]
```

Comment:
I don't want to have this stateful RSpecRunner object in a global state. By making a runner registry instead of a runner factory registry, you're going to expose the internal state of RRRSpec to public. This is not a good design.

Also, it's odd to instantiate a runner at loading time. It's better to minimize the computation at top-level.